enforce RFC1123 format

wanyunSu opened the pull request:
# Description
Fixes #538 

Enforce user defined session name match the RFC1123 format as required by k8s namespace

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future.
  (Indicate issue here: # (issue))

MRiganSUSX reviewed (changes requested):
Hi @wanyunSu,

Thank you for your changes. 

I have a few more comments:
- In the exceptions, if you are not modifying the parent init, you can simply use `pass`, so:
```
class DruncK8sPodAlreadyExists(DruncException):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        
-->

class DruncK8sPodAlreadyExists(DruncException):
    pass
```
- Thanks for mentioning this in the wiki. Could we also update the wiki to mention / explain how to run with a unified shell ? I assume that is what most users will want. 
- While this implementation achieves the correct format for the k8s pm (where it is absolutely required), the issue that this is solving https://github.com/DUNE-DAQ/drunc/issues/521 says that the limitation should be applied globally for all sessions in the base pm, regardless of the specific pm implemenation. If this sounds like too much of an extension of the PR, please open another issue and update the existing one mentioning that there is a case for k8s already for this, otherwise please move the implemenation to base pm ?

wanyunSu commented:
Thanks @MRiganSUSX for the review!

- I’ve simplified the k8s exception classes as suggested.
- I’ve updated the wiki with instructions on how to run with a unified shell.
- Regarding the limitation: my understanding is that this restriction originates from the k8s namespace requirement. I’m not sure it makes sense to extend it to the SSH PM, since it doesn’t have the same constraint. Could you clarify why we’d want this applied globally?

MRiganSUSX reviewed (approved):
lgtm
